htseq2multx.pl

```diff
@@ -0,0 +1,884 @@
+                        this wrapper emulates.  Supply twice to get a more
+                        verbose version of the barcode_splitter version that
+                        this wrapper emulates and the version of this wrapper.
+
+configuration arguments:
+  --fastq-multx PATH    Path to the fastq-multx installation.  Default is to use
+                        the PATH in the user's environment.
+  --debug               Supply this flag to output the fastq-multx command on
+                        standard error and not delete the temporary barcode
+                        file.
+END_HELP2
+
+    print($msg);
+  }
+
+sub getFastqMultxCommand
+  {
+    my $command =
+      "$fastq_multx -x -d 1 -B '$bcfilefqmx' -m $mismatches -M $mismatches";
+
+    $command .= ($barcodes_at_end ? ' -e' : ' -b');
+    $command .= ($check_deflines  ? " -v '$deflinesep'" : '');
+
+    #Add the input files and output file templates
+    my $templates_str = '';
+    foreach my $fqf_index (sort {indexesFirst($a,$b)}
+                           0..$#{$fastq_files})
+      {
+        $command       .= " '$fastq_files->[$fqf_index]'";
+        $templates_str .= " -o '" . getOutfileTemplate($fqf_index) . "'";
+      }
+    $command .= $templates_str;
+
+    return($command);
+  }
+
+#Runs the command and prints stdout/stderr.  Exits as barcode_splitter would.
+sub runFastqMultx
+  {
+    my $command = $_[0];
+
+    my $producer = new IO::Pipe::Producer();
+    my($stdout_handle,$stderr_handle) = $producer->getSystemProducer($command);
+
+    my $sel = new IO::Select;
+    $sel->add($stdout_handle,$stderr_handle);
+    my $stdout = '';
+    my $make_error_fatal = 0;
+    while(my @fhs = $sel->can_read())
+      {
+        foreach my $fh (@fhs)
+          {
+            my $line = <$fh>;
+            unless(defined($line))
+              {
+                $sel->remove($fh);
+                close($fh);
+                next;
+              }
+            if($fh == $stdout_handle)
+              {$stdout .= $line}
+            elsif($fh == $stderr_handle)
+              {
+                #Echo STDERR and exit non-0 if error is fatal
+                if(processSTDERR($line))
+                  {$make_error_fatal = 1}
+              }
+          }
+      }
+
+    if($? || $make_error_fatal)
+      {
+        print STDERR ("ERROR: fastq-multx command failed",
+                      ($? ? " with a non-zero exit code [$?]" .
+                       ($! =~ /./ ? " and error: $!" : '.') : ':'),
+                      "\n\t$command\n");
+        unless($stdout =~ /\t/)
+          {exit(1)}
+      }
+
+    print(fqmx2bcsStdout($stdout,scalar(@$idxread)));
+  }
+
+#This detects and converts files input via process substitution, and writes them
+#out to temporary files.  Assumes $gzip is true.
+sub processGZInfiles
+  {
+    my $fastq_files = $_[0];
+    my $prefix      = $_[1];
+    return() if(scalar(grep {$_ !~ /\.gz$/i} @$fastq_files) == 0);
```

Comment:
Don't need the `$_` here:

```suggestion
    return() if(scalar(grep {/\.gz$/i} @{ $fastq_files }) == 0);
```

Reply:
Removing `$_ !~` makes the filter do the opposite of how it was written.  I know you're trying to eliminate the cryptic `$_` variables, but in this instance, I do not know how to do that without adding cryptic stiff in or after the regex.